staff/judith-gongora/pre-course/bingo/bingo.js

```diff
@@ -0,0 +1,224 @@
+    }
+  }
+  if (comprobacion){
+    newCard() //Si se repite volvemos a generar un cartón, este no sirve
+  }else{ // Si no se repite
+    console.log("Aquí tienes tu cartón para empezar a jugar.\n")
+    showCard() //Mostramos el cartón
+  }
+}
+
+function randomNumber(){ //Funcioó que nos devuelve un número aleatorio entre 1 y 99
+  return Math.floor(Math.random() * 99) + 1
+}
+
+function showCard (){ //Funcion para mostrar el cartón
+  console.log("BINGO CARD")
+  var linea1 = [] //Creamos 1 array por cada linea, cada linea contendrá 5 números en mi caso
+  var linea2 = []
+  var linea3 = []
+  for (i = 0 ; i<5 ; i++){ //Recorremos la linea 1 
+      if (bingoCard[i].matched){ //Si en una linea hay un objeto con matched == true aparecerá una X
+        linea1.push("X")
+      }else{ //Sino mostrará el contenido de bingoCard[i].number
+        linea1.push(bingoCard[i].number)
+      }
+  }
+  for (i = 5 ; i<10 ; i++){
+      if (bingoCard[i].matched){
+        linea2.push("X")
+      }else{
+        linea2.push(bingoCard[i].number)
+      }
+  }
+  for (i = 10 ; i<15 ; i++){
+      if (bingoCard[i].matched){
+        linea3.push("X")
+      }else{
+        linea3.push(bingoCard[i].number)
+      }
+  }
+  console.log(linea1)//Mostramos los arrays con los numeros o X
+  console.log(linea2)
+  console.log(linea3)
+  if (primero){ //Si boleano primero es true preguntamos si quiere cambiar de cartón
+    var opcion = prompt("Si no te gustan los números que te han tocado puedes cambiar de cartón. \nQuieres cambiar?\n \nSi escribes cualquier cosa distina de Si o No, se continuará con el mismo.", "Si/No")
+    switch (opcion){ 
+      case "Si" ://Si la respuesta es Si volvemos a generar un nuevo cartón
+        console.clear()
+        newCard()//Llamada a la funcion newCard para proporcionar otro cartón
+        break
+      case "No" : //Si la respuesta es No:
+        primero = false //Cambiamos boleano primero para que no pregunte mas por el cambio
+        nuevoTurno() //Llamada a la funcion nuevoTurno para proseguir
+        break
+      default : //Si se ha escrito cualquier otra cosa, proseguimos
+        primero = false //Cambiamos boleano primero para que no pregunte mas por el cambio
+        console.log("No has escrito ni Si ni No, se prosigue con el mismo cartón.") 
+        nuevoTurno() //Llamada a la funcion nuevoTurno para proseguir
+        break
+    }
+  }else{ //Ya se ha aceptado el cartón por lo tanto no preguntamos por el cambio
+    nuevoTurno() //Llamada a la funcion nuevoTurno para proseguir
+  }
+}
+
+function nuevoTurno(){ //Funcion para preguntar al usuario si quiere otro número
+  var continuar = confirm("Quieres un número?")
+  if (continuar) { //Si quiere un nuevo numero
+    //console.clear()
+    turnos++ //Aumentamos variable turno
+    numeroNuevo() //Llamamos a la funcion
+  } else {
+    console.log("Gracias por jugar! Hasta la próxima.")
+  }
+}
+
+function numeroNuevo(){ //Funcion para generar un numero 
```

Comment:
Minimizar el numero de operaciones dentro de las funciones es bien. Puedes partir el código en nuevas funciones que hagan una operación en concreto y en un futuro pueden ser reutilizadas. Por ejemplo, en este caso partir la función numeroNuevo en otras como numeroUnico que se encarga de generar un numero único y comprobarNumero que se encarga de comprobar el numero único con los números del cartón.

Reply:
thanks for the advice!